Patterns/Serilog.md

```diff
@@ -30,6 +30,30 @@ _logger.LogWarning(caughtException, "Saving invoice failed.");
 
 This will save and include the stack trace of that `caughtException` in Seq, making debugging that unexpected `NullReferenceException` much easier because it's location can be better determined.
 
+### Create log scopes to correlate logging
+
+(Implementation can be found in Triple.Extensions.Logging)
+
+Use log scopes to correlate messages:
+
+```c#
+var invoiceId = "INV123";
+using (_logger.AddToScope(invoiceId).BeginScope("Invoice processing"))
+{
+    _invoiceService.DoStuff(invoiceId);
+}
+```
+
+This will create a log scope that will, until it is disposed, add the `invoiceId` property with value `"INV123"` to every log message that is created within the scope. The logs inside the Invoice Service will also have the `invoiceId` property added. Next to that, the `BeginScope` method will log an `"Invoice processing started"`-message at the start of the scope and `"Invoice processing finished"` once it is disposed. 
+
+### Correlate logging between requests
+
+Correlating log messages using unique identifiers makes debugging interactions between services less challenging. Including the TraceIdentifier that exists on the HttpContext in a very global log scope makes correlating log messages for an entire request possible. Sending the TraceIdentifier in a header (like `X-Trace`) to other services which log that header makes it possible to correlate a request and it subsequent downstream requests.
```

Comment:
At Mijn Zaak we now add the SpanId as well, in which you add the requesting service as well. In addition to that we add the environment in that Span (can be helpful in case of misconfigurations between services).

X-MZ-TraceId -> 8000000e-0001-fd00-b63f-84710c7967bb
X-MZ-SpanId -> MZ-Development

Optionally a Parent ID can also be used in the second service, to save the previous Span ID in. But we have not implemented that. Some more reading on this with pretty pictures:
https://blog.sentry.io/2021/08/12/distributed-tracing-101-for-full-stack-developers